### ISSUE TITLE
getEventResourceField Javascript error when move event

### ISSUE THREAD

**Issue by martinlaregina** [Edofre/yii2-fullcalendar-scheduler#8]
Hi again,

Thanks for your reply. Its work now, but with js error. I use "FullCalendar Scheduler v1.4.0" so when i move a event like screenshot attached i see this error.

I made a backup and before update its work with out js error. After update via composer.json
"edofre/yii2-fullcalendar-scheduler": "=1.1.4", generate this error. 

I can send you the Yii code if you cant reproduce this error. I put "alert" on every event and dont show any alert to debug the error. 

Thanks a lot,
![javascript_error](https://cloud.githubusercontent.com/assets/11611304/21318727/bb08291e-c60a-11e6-84c6-9881b0dcec65.jpg)

**Comment by Edofre** [comment]
What are you trying to do that generates the error? Can you share the code that generates the schedular/events/resources?

I can't find the function: "getEventResourceField" in the documentation.

**Comment by martinlaregina** [comment]
Hi again,

Here you have the resource.json response. I cut a little the response because are very large. But i think is not a problem or response, because same code before update works well with out error, and after update show this error of JS. Even with the error, the record are updated. Imagine you have a interview today at 12:00 and want to move it from 12:00 to 13:00 just click on an event and move it to 13:00. The error produce it before fire any event of you code, its fired when starting to move the event to another place. When i drop the event, your code call "update url" and event is updated well, but i dont know why when i moving the event, show this error.

Tks a lot and sorry for my poor english.

`
[{"id":13,"title":"Analia","eventColor":null,"eventBackgroundColor":"#423824","eventBorderColor":"#423824","eventTextColor":null,"eventClassName":null,"children":[],"parentId":null,"parent":null},{"id":11,"title":"Catalina","eventColor":null,"eventBackgroundColor":"#32ba79","eventBorderColor":"#32ba79","eventTextColor":null,"eventClassName":null,"children":[],"parentId":null,"parent":null},
`

and event.json response

`
[{"id":203,"resourceId":11,"title":"Anestesia","allDay":false,"start":"2016-10-07 11:00:00","end":"2016-10-07 14:00:00","url":null,"className":null,"editable":true,"startEditable":true,"durationEditable":true,"rendering":null,"overlap":true,"constraint":null,"source":null,"color":null,"backgroundColor":null,"borderColor":null,"textColor":null},{"id":289,"resourceId":13,"title":"Anestesia","allDay":false,"start":"2016-10-10 10:00:00","end":"2016-10-10 14:00:00","url":null,"className":null,"editable":true,"startEditable":true,"durationEditable":true,"rendering":null,"overlap":true,"constraint":null,"source":null,"color":null,"backgroundColor":null,"borderColor":null,"textColor":null}]
`

**Comment by martinlaregina** [comment]
Here you have Js code generated.
`<script type="text/javascript">jQuery(document).ready(function () {
jQuery('#calendar').fullCalendar({"loading":function(isLoading, view ) {
			jQuery('#calendar').find('.fc-loading').toggle(isLoading);
        },"events":"/scheduler/events","resources":"/scheduler/resources","schedulerLicenseKey":"CC-Attribution-NonCommercial-NoDerivatives","now":"2016-12-19","displayEventTime":true,"droppable":true,"displayEventEnd":true,"eventDurationEditable":true,"scrollTime":"09:00","defaultView":"timelineDay","timeFormat":"HH:mm","resourceAreaWidth":"15%","views":{"timelineDay":{"slotDuration":"00:30","snapDuration":"00:10","slotLabelFormat":["HH:mm"]}},"resourceLabelText":"Empleados","eventOverlap":true,"resourceColumns":[{"render":
                  function(resource, el) {
                    el.css('color', '#696969');
                    //el.css('font-weight', 'bold');
                    el.html('<i class="fa fa-user" style="color:'+resource.eventBackgroundColor+'" aria-hidden="true"></i>&nbsp; '+resource.title);
                  }
                  }],"drop":
              function(date, jsEvent, ui, resourceId) {
                var employeeid = resourceId;
                var rolid      = $.trim($(this).attr('data-rolid'));
                var date_start = date.format();
                $.ajax({
                  url: '/scheduler/create',
                  data: 'rolid='+ rolid +'&employeeid='+ employeeid +'&date_start='+ date_start,
                  type: 'POST',
                  beforeSend: loading('show'),
                }).done(function(data){
                  if (data.success)
                  {
                    $.notify(data.message,{type:'success', delay:1});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  } else {
                    $.notify(data.message,{type:'error', delay:10});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  }
                }).always(function(){
                  loading('hide');
                }).fail(function(response){
                  $.notify(response,{type:'error', delay:10});
                  $('#calendar').fullCalendar('removeEvents');
                  $('#calendar').fullCalendar('refetchEvents');
                });
              }
            ,"eventReceive":
              function(event) { // called when a proper external event is dropped
                event.editable = true;
                event.resourceEditable = true;
                event.eventDurationEditable = true;
              }
            ,"eventClick":
              function(calEvent, jsEvent, view) {
                var eventid = calEvent.id;
                if (bootbox.confirm('Está seguro que desea eliminar el evento ?', function(result) {
                  if (result){
                    $.ajax({
                      url: '/scheduler/delete/?id=' + eventid,
                      data: 'id='+ eventid,
                      type: 'POST',
                      async: false,
                      beforeSend: function(){
                        loading('show');
                      },
                    }).done(function(data){
                      if (data.success)
                      {
                        $('#calendar').fullCalendar('removeEvents', eventid);
                        $.notify(data.message,{type:'success', delay:1});
                        $('#calendar').fullCalendar('removeEvents');
                        $('#calendar').fullCalendar('refetchEvents');
                      } else {
                        $.notify(data.message,{type:'error', delay:10});
                      }
                    }).always(function(){
                      loading('hide');
                    }).fail(function(response){
                      $.notify(response,{type:'error', delay:10});
                    });
                  }
                }));
              }
            ,"eventDrop":
              function(event) { // called when an event (already on the calendar) is moved
                var schedulerid = event.id;
                var employeeid  = event.resourceId;
                var date_start  = event.start.format();
                var date_stop   = event.end.format();
                $.ajax({
                  url: '/scheduler/update/?id=' + schedulerid,
                  data: 'id='+ schedulerid +'&employeeid='+ employeeid +'&date_start='+ date_start +'&date_stop='+ date_stop,
                  type: 'POST',
                  beforeSend: loading('show'),
                }).done(function(data){
                  if (data.success)
                  {
                    $.notify(data.message,{type:'success', delay:1});
                  } else {
                    $.notify(data.message,{type:'error', delay:10});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  }
                }).always(function(){
                  loading('hide');
                }).fail(function(response){
                  $.notify(response,{type:'error', delay:10});
                  $('#calendar').fullCalendar('removeEvents');
                  $('#calendar').fullCalendar('refetchEvents');
                });
              }
            ,"eventResize":
              function(event, delta, revertFunc) {
                var schedulerid = event.id;
                var employeeid  = event.resourceId;
                var date_start  = event.start.format();
                var date_stop   = event.end.format();
                $.ajax({
                  url: '/scheduler/update/?id=' + schedulerid,
                  data: 'id='+ schedulerid +'&employeeid='+ employeeid +'&date_start='+ date_start +'&date_stop='+ date_stop,
                  type: 'POST',
                  beforeSend: loading('show'),
                }).done(function(data){
                  if (data.success)
                  {
                    $.notify(data.message,{type:'success', delay:1});
                  } else {
                    $.notify(data.message,{type:'error', delay:10});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  }
                }).always(function(){
                  loading('hide');
                }).fail(function(response){
                  $.notify(response,{type:'error', delay:10});
                  $('#calendar').fullCalendar('removeEvents');
                  $('#calendar').fullCalendar('refetchEvents');
                });
              }
            ,"header":{"left":"today prev,next","center":"title","right":"timelineDay,agendaWeek,month,listDay"}});
});</script>`

**Comment by martinlaregina** [comment]
I was debugging and i see the version "FullCalendar v3.0.1" is not working well, and with version "FullCalendar v2.7.2" all works fine.

**Comment by Edofre** [comment]
Can you share the PHP code you use to generate the widget?

**Comment by martinlaregina** [comment]
![screenshot_1](https://cloud.githubusercontent.com/assets/11611304/21322493/a51e49e4-c619-11e6-9ccf-9c4c181ac743.jpg)
Sure, here you have...
`<?php

use yii\helpers\Html;

use kartik\growl\GrowlAsset;
use kartik\base\AnimateAsset;

GrowlAsset::register($this);
AnimateAsset::register($this);

/* @var $this yii\web\View */

$this->title = Yii::t('scheduler', 'Planificación');
$this->params['breadcrumbs'][] = ['label' => Yii::t('scheduler', 'Planificación'), 'url' => ['index']];

$this->registerJs("

  /* initialize the external events
  -----------------------------------------------------------------*/
  var loading = function (data) {
    if (data=='show') {
      Pace.restart();
      $('#loading_scheduler').removeClass('hide');
    } else {
      $('#loading_scheduler').addClass('hide');
    }
  }
  /* initialize the external events
  -----------------------------------------------------------------*/
  $('#external-events .fc-event').each(function() {
    // store data so the calendar knows to render an event upon drop
    $(this).data('event', {
      title: $.trim($(this).text()), // use the element's text as the event title
      stick: true // maintain when user navigates (see docs on the renderEvent method)
    });
    // make the event draggable using jQuery UI
    $(this).draggable({
      zIndex: 999,
      revert: true,      // will cause the event to go back to its
      revertDuration: 0  //  original position after the drag
    });
  });
", \yii\web\View::POS_END);

?>

    <div class="row">
        <div class="col-xs-12 col-sm-12">
            <div class="box box-warning">
                <div class="box-header with-border">
                    <h3 class="box-title">
                        <?= Yii::t('app', 'Filtros') ?>
                    </h3>
                    <img src="<?= Yii::getAlias('@web'); ?>/img/loading.gif" style="margin-left:10px;margin-top:-5px" alt="" id="loading_scheduler" class="hide" />
                </div>
                <div class="box-body">
                    <div class="row">
                        <div class="col-sm-2">
                            <div id="external-events">
                                <h4>
                                    <?= Yii::t('rol', 'Roles') ?>
                                </h4>
                                <?php
              foreach ($dataProvider->models as $model) {
                echo '<div class="fc-event" data-rolid="'.$model->id.'" data-duration="'.date('i\:s', mktime(0,0,Yii::$app->params['duration'])).'">'.$model->name.'</div>';
              }
              ?>
                            </div>
                        </div>
                        <div class="col-sm-10">
                            <?= \edofre\fullcalendarscheduler\FullcalendarScheduler::widget([
            'header'        => [
              'left'   => 'today prev,next',
              'center' => 'title',
              'right'  => 'timelineDay,agendaWeek,month,listDay',
            ],
            'clientOptions' => [
              'schedulerLicenseKey' => 'CC-Attribution-NonCommercial-NoDerivatives',
              'now'                 => date('Y-m-d'),
              //'aspectRatio'       => 3.8,
              'displayEventTime'    => true,
              'droppable'           => true,
              'displayEventEnd'     => true,
              /*'dragRevertDuration'  => 0,*/
              //'durationEditable'    => true,
              //'resourceEditable'    => true,
              'eventDurationEditable' => true,
              'scrollTime'          => '09:00',
              'defaultView'         => 'timelineDay',
              'timeFormat'          => 'HH:mm',
              'resourceAreaWidth'   => '15%',
              'views'               => [
                'timelineDay'       => [
                  'slotDuration'    => '00:30',
                  'snapDuration'    => '00:10',
                  'slotLabelFormat' => ['HH:mm']
                ],
              ],
              'resourceLabelText' => Yii::t('employee', 'Empleados'),
              'eventOverlap'      => true,
              'resources'         => \yii\helpers\Url::to(['scheduler/resources']),
              'events'            => \yii\helpers\Url::to(['scheduler/events']),
              'resourceColumns'   => [
                ['render' => new \yii\web\JsExpression("
                  function(resource, el) {
                    el.css('color', '#696969');
                    //el.css('font-weight', 'bold');
                    el.html('<i class=\"fa fa-user\" style=\"color:'+resource.eventBackgroundColor+'\" aria-hidden=\"true\"></i>&nbsp; '+resource.title);
                  }
                  "),
                ],
              ],
            'drop' => new \yii\web\JsExpression("
              function(date, jsEvent, ui, resourceId) {
                var employeeid = resourceId;
                var rolid      = $.trim($(this).attr('data-rolid'));
                var date_start = date.format();
                $.ajax({
                  url: '".\yii\helpers\Url::to(['scheduler/create'])."',
                  data: 'rolid='+ rolid +'&employeeid='+ employeeid +'&date_start='+ date_start,
                  type: 'POST',
                  beforeSend: loading('show'),
                }).done(function(data){
                  if (data.success)
                  {
                    $.notify(data.message,{type:'success', delay:1});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  } else {
                    $.notify(data.message,{type:'error', delay:10});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  }
                }).always(function(){
                  loading('hide');
                }).fail(function(response){
                  $.notify(response,{type:'error', delay:10});
                  $('#calendar').fullCalendar('removeEvents');
                  $('#calendar').fullCalendar('refetchEvents');
                });
              }
            "),
            'eventReceive' => new \yii\web\JsExpression("
              function(event) { // called when a proper external event is dropped
                event.editable = true;
                event.resourceEditable = true;
                event.eventDurationEditable = true;
              }
            "),
            'eventClick' => new \yii\web\JsExpression("
              function(calEvent, jsEvent, view) {
                var eventid = calEvent.id;
                if (bootbox.confirm('".Yii::t('scheduler', 'Está seguro que desea eliminar el evento ?')."', function(result) {
                  if (result){
                    $.ajax({
                      url: '".\yii\helpers\Url::to(['scheduler/delete'])."/?id=' + eventid,
                      data: 'id='+ eventid,
                      type: 'POST',
                      async: false,
                      beforeSend: function(){
                        loading('show');
                      },
                    }).done(function(data){
                      if (data.success)
                      {
                        $('#calendar').fullCalendar('removeEvents', eventid);
                        $.notify(data.message,{type:'success', delay:1});
                        $('#calendar').fullCalendar('removeEvents');
                        $('#calendar').fullCalendar('refetchEvents');
                      } else {
                        $.notify(data.message,{type:'error', delay:10});
                      }
                    }).always(function(){
                      loading('hide');
                    }).fail(function(response){
                      $.notify(response,{type:'error', delay:10});
                    });
                  }
                }));
              }
            "),
            'eventDrop'    => new \yii\web\JsExpression("
              function(event) { // called when an event (already on the calendar) is moved
                var schedulerid = event.id;
                var employeeid  = event.resourceId;
                var date_start  = event.start.format();
                var date_stop   = event.end.format();
                $.ajax({
                  url: '".\yii\helpers\Url::to(['scheduler/update'])."/?id=' + schedulerid,
                  data: 'id='+ schedulerid +'&employeeid='+ employeeid +'&date_start='+ date_start +'&date_stop='+ date_stop,
                  type: 'POST',
                  beforeSend: loading('show'),
                }).done(function(data){
                  if (data.success)
                  {
                    $.notify(data.message,{type:'success', delay:1});
                  } else {
                    $.notify(data.message,{type:'error', delay:10});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  }
                }).always(function(){
                  loading('hide');
                }).fail(function(response){
                  $.notify(response,{type:'error', delay:10});
                  $('#calendar').fullCalendar('removeEvents');
                  $('#calendar').fullCalendar('refetchEvents');
                });
              }
            "),
            'eventResize' => new \yii\web\JsExpression("
              function(event, delta, revertFunc) {
                var schedulerid = event.id;
                var employeeid  = event.resourceId;
                var date_start  = event.start.format();
                var date_stop   = event.end.format();
                $.ajax({
                  url: '".\yii\helpers\Url::to(['scheduler/update'])."/?id=' + schedulerid,
                  data: 'id='+ schedulerid +'&employeeid='+ employeeid +'&date_start='+ date_start +'&date_stop='+ date_stop,
                  type: 'POST',
                  beforeSend: loading('show'),
                }).done(function(data){
                  if (data.success)
                  {
                    $.notify(data.message,{type:'success', delay:1});
                  } else {
                    $.notify(data.message,{type:'error', delay:10});
                    $('#calendar').fullCalendar('removeEvents');
                    $('#calendar').fullCalendar('refetchEvents');
                  }
                }).always(function(){
                  loading('hide');
                }).fail(function(response){
                  $.notify(response,{type:'error', delay:10});
                  $('#calendar').fullCalendar('removeEvents');
                  $('#calendar').fullCalendar('refetchEvents');
                });
              }
            "),
            ],
          ]);
          ?>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <?php $this->registerCss('
  #wrap {
    width: 1100px;
    margin: 0 auto;
  }
  #external-events {
    float: left;
    width: 100%;
    padding: 0 10px;
    border: 1px solid #ccc;
    background: #eee;
    text-align: left;
    margin-bottom: 10px;
  }
  #external-events h4 {
    font-size: 16px;
    margin-top: 0;
    padding-top: 1em;
  }
  #external-events .fc-event {
    margin: 10px 0;
    cursor: pointer;
    line-height: 1.8em;
    padding-left: 5px;
    background: #f4f4f4;
    color: #000;
    border: 1px solid #ddd;
  }
  .fc-event-container:hover {
    cursor: pointer!important;
  }
');
?>`

**Comment by Edofre** [comment]
Hm, I can't find out where the error comes from at the moment. If I were you I would look at how you're generating the events and resources. If I load other resources & events I don't get the errors. When I'm back from the gym I will look again. (Be back in around 3 hours.)

**Comment by Edofre** [comment]
@martinlaregina 

I've updated to v1.1.5. I removed the default attributes from the Event model and the JSON response of an Event model will now remove the unused attributes.

The problem was that in an Event response it would return attributes with null. The updated scheduler does not ignore these values like the old scheduler and that's why we got the resourceId errors. 

Thank you for bringing this to my attention, I hope it's working properly for you now 👍
